src/client/Microsoft.Identity.Client/Instance/Oidc/OidcRetrieverWithCache.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Client.OAuth2;
+
+namespace Microsoft.Identity.Client.Instance.Oidc
+{
+    internal static class OidcRetrieverWithCache
+    {
+        private static readonly ConcurrentDictionary<string, OidcMetadata> s_cache = new();
+        private static readonly SemaphoreSlim s_lockOidcRetrieval = new SemaphoreSlim(1);
+
+        internal const string OpenIdConfigurationEndpointSuffix = ".well-known/openid-configuration";
```

Comment:
Seems like this is already available from Constants.cs. No need to have a duplicate.